arrow-libs/core/arrow-core/src/commonMain/kotlin/arrow/core/Eval.kt

```diff
@@ -233,7 +233,7 @@ sealed class Eval<out A> {
       is FlatMap<A> -> object : FlatMap<B>() {
         override fun <S> start(): Eval<S> = (this@Eval).start()
 
-        @IgnoreJRERequirement
+        // @IgnoreJRERequirement
```

Comment:
It seems this annotation is not needed anymore. AnimalSniffer seems to pass on this code.

Reply:
Oh, I'm going to review AnimalSniffer configuration because maybe it must be updated with the last change.

Reply:
Alright, if we need it back we can turn it into a `expect/actual` annotation where native and JS have no-op annotations.